clients/geth/specular/rollup/services/validator/validator.go

```diff
@@ -42,141 +42,58 @@ func New(eth services.Backend, proofBackend proof.Backend, cfg *services.Config,
 	}
 	v := &Validator{
 		BaseService:          base,
-		batchCh:              make(chan *rollupTypes.TxBatch, 4096),
+		newBatchCh:           make(chan struct{}, 4096),
 		challengeCh:          make(chan *challengeCtx),
-		challengeResoutionCh: make(chan struct{}),
+		challengeResoutionCh: make(chan *rollupTypes.Assertion),
 	}
 	return v, nil
 }
 
-// commitBlocks executes and commits sequenced blocks to local blockchain
-// It returns the afterwards state hash and total gas used for these blocks
-// TODO: this function shares a lot of codes with Batcher
-// TODO: use StateProcessor::Process() instead
-func (v *Validator) commitBlocks(blocks []*rollupTypes.SequenceBlock) (common.Hash, *big.Int, error) {
-	var targetVmHash common.Hash
-	targetGasUsed := new(big.Int)
-	chainConfig := v.Chain.Config()
-	parent := v.Chain.CurrentBlock()
-	if parent == nil {
-		return common.Hash{}, nil, fmt.Errorf("missing parent")
-	}
-	num := parent.Number()
-	if num.Uint64() != blocks[0].BlockNumber-1 {
-		return common.Hash{}, nil, fmt.Errorf("validator unsynced")
-	}
-	state, err := v.Chain.StateAt(parent.Root())
-	// TODO: in newest version of geth, below codes are removed
-	if err != nil {
-		// Note since the sealing block can be created upon the arbitrary parent
-		// block, but the state of parent block may already be pruned, so the necessary
-		// state recovery is needed here in the future.
-		//
-		// The maximum acceptable reorg depth can be limited by the finalised block
-		// somehow. TODO(rjl493456442) fix the hard-coded number here later.
-		state, err = v.Eth.StateAtBlock(parent, 1024, nil, false, false)
-		log.Warn("Recovered mining state", "root", parent.Root(), "err", err)
-	}
-	if err != nil {
-		return common.Hash{}, nil, err
-	}
-	state.StartPrefetcher("validator")
-	defer state.StopPrefetcher()
-
-	for _, sblock := range blocks {
-		header := &types.Header{
-			ParentHash: parent.Hash(),
-			Number:     new(big.Int).SetUint64(sblock.BlockNumber),
-			GasLimit:   core.CalcGasLimit(parent.GasLimit(), ethconfig.Defaults.Miner.GasCeil), // TODO: this may cause problem
-			Time:       sblock.Timestamp,
-			Coinbase:   v.Config.SequencerAddr,
-			Difficulty: common.Big1, // Fake difficulty. Avoid use 0 here because it means the merge happened
-		}
-		gasPool := new(core.GasPool).AddGas(header.GasLimit)
-		var receipts []*types.Receipt
-		for idx, tx := range sblock.Txs {
-			state.Prepare(tx.Hash(), idx)
-			receipt, err := core.ApplyTransaction(chainConfig, v.Chain, &v.Config.SequencerAddr, gasPool, state, header, tx, &header.GasUsed, *v.Chain.GetVMConfig())
-			if err != nil {
-				return common.Hash{}, nil, err
-			}
-			receipts = append(receipts, receipt)
+// This function will try to validate a pending assertion
+func (v *Validator) tryVaidateAssertion(confirmedAssertion, assertion *rollupTypes.Assertion) error {
```

Comment:
```suggestion
func (v *Validator) tryValidateAssertion(confirmedAssertion, assertion *rollupTypes.Assertion) error {
```
also nit maybe we should call this `locallyConfirmedAssertion` (to disambiguate from confirmation in the L1 sense which implies finality)